src/transformers/models/blip_2/modeling_blip_2.py

```diff
@@ -1203,8 +1204,45 @@ def __init__(self, config: Blip2Config):
         # Initialize weights and apply final processing
         self.post_init()
 
-    def get_input_embeddings(self) -> nn.Module:
-        return self.vision_model.embeddings.patch_embedding
+    def get_input_embeddings(self):
+        return self.language_model.get_input_embeddings()
+
+    def set_input_embeddings(self, value):
+        self.language_model.set_input_embeddings(value)
+
+    def set_output_embeddings(self, new_embeddings):
+        self.language_model.set_output_embeddings(new_embeddings)
+
+    def get_output_embeddings(self) -> nn.Module:
+        return self.language_model.get_output_embeddings()
+
+    def get_encoder(self):
+        return self.language_model.get_encoder()
+
+    def get_decoder(self):
+        return self.language_model.get_decoder()
+
+    def _tie_weights(self):
+        if not self.config.use_decoder_only_language_model:
+            self.language_model.encoder.embed_tokens = self.language_model.shared
+            self.language_model.decoder.embed_tokens = self.language_model.shared
+
+    def _preprocess_accelerate(self):
+        r"""
+        Some pre-processing hacks to make the model `accelerate` compatible. Check
+        https://github.com/huggingface/transformers/pull/21707 for more details.
+        """
+        hf_device_map = self.hf_device_map
+
+        if len(hf_device_map) > 1 and "language_model" not in hf_device_map and torch.cuda.device_count() > 1:
+            # warn users about unexpected behavior when using multi-GPU + BLIP-2 + `accelerate`.
+            logger.warning(
+                "The `language_model` is not in the `hf_device_map` dictionary and you are running your script in a multi-GPU environment. "
+                " this may lead to unexpected behavior when using `accelerate`. Please pass a `device_map` that contains `language_model` to remove this warning"
+            )
```

Comment:
This should never happen as Accelerate will error if you pass along a device_map that is incomplete.

Reply:
In fact, what I mean here is to have a device map that contains the exact key `language_model` , happy to edit the error message a bit if you want

Reply:
@younesbelkada 
Can we make this warning more scalable towards larger models? e.g., when trying to use `Salesforce/blip2-flan-t5-xxl`, the whole `language_model` may not fit into consume-level GPUs (e.g., on my 16GB V100). 

I believe adding the link to https://github.com/huggingface/blog/blob/main/accelerate-large-models.md  and guide it to make it work on larger models would be quite helpful.

FYI, to scale it to larger models, I follow what is instructed in https://github.com/huggingface/blog/blob/main/accelerate-large-models.md i.e.,

For OPT-* models, 
```
with init_empty_weights():
    model = Blip2ForConditionalGeneration(config)
    device_map = infer_auto_device_map(model, no_split_module_classes=["OPTDecoderLayer"])
device_map['language_model.lm_head'] = device_map["language_model.model.decoder.embed_tokens"]  # to make the generated tokens and input_ids to be on the same device
```

For Flan-T5-* models, 
```
with init_empty_weights():
    model = Blip2ForConditionalGeneration(config)
    device_map = infer_auto_device_map(model, no_split_module_classes=["T5Block"])
device_map['language_model.lm_head'] = device_map["language_model.decoder.embed_tokens"]  # to make the genearted tokens and input_ids to be on the same device
```

Again, thanks a lot for this PR!